client/my-sites/patterns/home/index.tsx

```diff
@@ -16,10 +18,28 @@ import ImgLayout from './images/layout.svg';
 import ImgPattern from './images/pattern.svg';
 import ImgResponsive from './images/responsive.svg';
 import ImgStyle from './images/style.svg';
-import type { PatternGalleryFC } from 'calypso/my-sites/patterns/types';
+import type { Pattern, PatternGalleryFC } from 'calypso/my-sites/patterns/types';
 
 import './style.scss';
 
+function filterPatternsByQuery( patterns: Pattern[], searchTerm: string ) {
+	if ( ! searchTerm ) {
+		return [];
```

Comment:
I would have expected that function to return the whole list if the search term is empty:

```suggestion
		return patterns;
```

And this would allow us to simplify that calling code:

```javascript
const patternSearchResults = filterPatternsByQuery( patterns, searchTerm );
const patternsToRender = searchTerm ? patternSearchResults : patterns;
```

But maybe I've missed some edge case?